openmdao/components/interp_util/interp_lagrange3.py

```diff
@@ -909,3 +909,795 @@ def compute_coeffs_vectorized(self, idx, dtype):
+        Cache of all computed coefficients when running vectorized.
+    """
+
+    def __init__(self, grid, values, interp, **kwargs):
+        """
+        Initialize table and subtables.
+        """
+        super().__init__(grid, values, interp)
+        self.coeffs = {}
+        self.vec_coeff = None
+        self.k = 4
+        self.dim = 2
+        self.last_index = [0] * self.dim
+        self._name = '2D-lagrange3'
+        self._vectorized = False
+
+    def vectorized(self, x):
+        """
+        Return whether this table will be run vectorized for the given requested input.
+
+        Parameters
+        ----------
+        x : float
+            Value of new independent to interpolate.
+
+        Returns
+        -------
+        bool
+            Returns True if this table can be run vectorized.
+        """
+        # If we only have 1 point, use the non-vectorized implementation, which has faster
+        # bracketing than the numpy version.
+        return x.shape[0] > 1
+
+    def interpolate(self, x, idx):
+        """
+        Compute the interpolated value.
+
+        Parameters
+        ----------
+        x : ndarray
+            The coordinates to sample the gridded data at. First array element is the point to
+            interpolate here. Remaining elements are interpolated on sub tables.
+        idx : int
+            Interval index for x.
+
+        Returns
+        -------
+        ndarray
+            Interpolated values.
+        ndarray
+            Derivative of interpolated values with respect to this independent and child
+            independents.
+        ndarray
+            Derivative of interpolated values with respect to values for this and subsequent table
+            dimensions.
+        ndarray
+            Derivative of interpolated values with respect to grid for this and subsequent table
+            dimensions.
+        """
+        grid = self.grid
+        i_x, i_y = idx
+
+        # Extrapolation
+        # Shift if we don't have 2 points on each side.
+        n = len(grid[0])
+        if i_x > n - 3:
+            i_x = n - 3
+        elif i_x < 1:
+            i_x = 1
+
+        n = len(grid[1])
+        if i_y > n - 3:
+            i_y = n - 3
+        elif i_y < 1:
+            i_y = 1
+
+        idx = (i_x, i_y)
+
+        # Complex Step
+        if self.values.dtype == complex:
+            dtype = self.values.dtype
+        else:
+            dtype = x.dtype
+
+        if idx not in self.coeffs:
+            self.coeffs[idx] = self.compute_coeffs(idx, dtype)
+        a = self.coeffs[idx]
+
+        x, y = x
+
+        # Taking powers of the "deltas" instead of the actual table inputs eliminates numerical
+        # problems that arise from the scaling of each axis.
+        x = x - grid[0][i_x - 1]
+        y = y - grid[1][i_y - 1]
+
+        # Compute interpolated value using the 16 coefficients.
+
+        xx = np.array([1.0, x, x * x, x * x * x], dtype=dtype)
```

Comment:
could save a tiny bit (a few multiplies) here by having x2=x*x and y2=y*y and using those when you define xx, yy, dx, and dy.